app/controllers/invites_controller.rb

```diff
@@ -23,84 +20,56 @@ def show
         I18n.t('invite.expired', base_url: Discourse.base_url)
-      elsif invite.present? && invite.redeemed?
+      elsif invite&.redeemed?
         I18n.t('invite.not_found_template', site_name: SiteSetting.title, base_url: Discourse.base_url)
       else
         I18n.t('invite.not_found', base_url: Discourse.base_url)
       end
+
       render layout: 'no_ember'
     end
   end
 
-  def perform_accept_invitation
-    params.require(:id)
-    params.permit(:email, :username, :name, :password, :timezone, user_custom_fields: {})
-    invite = Invite.find_by(invite_key: params[:id])
-
-    if invite.present?
-      begin
-        user = if invite.is_invite_link?
-          invite.redeem_invite_link(email: params[:email], username: params[:username], name: params[:name], password: params[:password], user_custom_fields: params[:user_custom_fields], ip_address: request.remote_ip)
-        else
-          invite.redeem(username: params[:username], name: params[:name], password: params[:password], user_custom_fields: params[:user_custom_fields], ip_address: request.remote_ip)
-        end
-
-        if user.present?
-          log_on_user(user) if user.active?
-          user.update_timezone_if_missing(params[:timezone])
-          post_process_invite(user)
-          response = { success: true }
-        else
-          response = { success: false, message: I18n.t('invite.not_found_json') }
-        end
-
-        if user.present? && user.active?
-          topic = invite.topics.first
-          response[:redirect_to] = topic.present? ? path("#{topic.relative_url}") : path("/")
-        elsif user.present?
-          response[:message] = I18n.t('invite.confirm_email')
-        end
+  def create
+    guardian.ensure_can_send_invite_links!(current_user) if params[:email].blank? || params[:skip_email].present?
 
-        render json: response
-      rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotSaved => e
-        render json: {
-          success: false,
-          errors: e.record&.errors&.to_hash || {},
-          message: I18n.t('invite.error_message')
-        }
-      rescue Invite::UserExists => e
-        render json: { success: false, message: [e.message] }
-      end
-    else
-      render json: { success: false, message: I18n.t('invite.not_found_json') }
+    if params[:email].present? && Invite.exists?(email: params[:email])
+      return render json: failed_json, status: 422
     end
-  end
 
-  def create
-    params.require(:email)
+    if params[:topic_id].present?
+      topic = Topic.find_by(id: params[:topic_id])
+      guardian.ensure_can_invite_to!(topic)
+    end
 
-    groups = Group.lookup_groups(
-      group_ids: params[:group_ids],
-      group_names: params[:group_names]
-    )
+    if params[:group_ids].present? || params[:group_names].present?
+      groups = Group.lookup_groups(group_ids: params[:group_ids], group_names: params[:group_names])
+    end
 
     guardian.ensure_can_invite_to_forum!(groups)
```

Comment:
Should this be moved into conditional above? Otherwise `groups` is not a defined variable if `params[:group_ids]` or `params[:group_names]` is not present.

Reply:
Even if that branch is never evaluated, `groups` is defined and it is `nil`. `guardian.ensure_can_invite_to_forum!` does more checks, some which are not related to `groups` and this is expectd.

Reply:
Ahh icic. I didn't know that if a variable is defined in a `if` statement it will evaluate to `nil` if the conditional is not met.